### DISCUSSION TITLE
Accessing GitHub has become a difficult task in China

### DISCUSSION THREAD

**Discussion by GavZheng** [community/community#82176] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Dear GitHub team,
I hope everything is well with you. I am writing to bring to your attention a problem that many users in China are facing - the instability of GitHub access.
For many of us in China, accessing GitHub is highly inconsistent. Sometimes the connection is fast, sometimes it takes over 10 seconds, and sometimes it is completely unreachable. I myself have attempted various methods from the Chinese internet, including modifying DNS settings, but these efforts proved mostly ineffective. Whether or not I can access GitHub successfully is entirely dependent on luck, which has caused me a lot of distress. Not to mention, the majority of China's 7.75 million users face similar issues.
It's worth noting that I am just a student interested in programming. Although my problem is already troublesome enough, I am well aware that compared to more professional developers, their problems will only multiply. They rely on GitHub to complete important tasks like project collaboration and code hosting. This instability undoubtedly brings tremendous inconvenience and challenges to their development work. This makes me realize that I should remind you to help Chinese developers solve this problem.
Additionally, I would like to remind you that using VPN and other proxy tools is illegal in mainland China, so we cannot use proxies to accelerate GitHub access. Therefore, I hope you can take this issue seriously and take prompt action to address it.
I believe that with your support and possible solutions, we can improve this situation. I strongly recommend that the GitHub team considers increasing server nodes in China or exploring other solutions to address this long-standing issue.
I hope this message, which carries the voices of most Chinese developers, will not fall on deaf ears. I sincerely look forward to your positive response and solving this long-standing problem for us.
Best regards!
Cat Bayi

亲爱的GitHub团队，
我希望你们一切都好。我写这封邮件是为了引起你们的注意，许多中国的用户都面临一个问题——GitHub访问的不稳定性。
对于我们许多中国用户来说，访问GitHub的连接非常不稳定。有时连接速度很快，有时需要超过10秒，有时甚至完全无法访问。我自己也尝试了各种来自中国互联网的方法，包括修改DNS设置，但这些努力大多无效。每次尝试访问GitHub时，成功与否完全取决于运气，这给我带来了很大的困扰。更不用说，中国750万用户中的大多数都面临着类似的问题。
值得注意的是，我只是一个对编程感兴趣的中学生。虽然我的问题已经很麻烦了，但我很清楚，与更专业的开发人员相比，他们的问题只会越来越多。他们依赖GitHub完成项目协作和代码托管等重要任务，这种不稳定性无疑给他们的开发工作带来了巨大的不便和挑战。这让我意识到我应该提醒你们帮助中国开发者解决这个问题。
此外，我想提醒你们，在中国大陆使用VPN等代理工具是违法的，我们无法使用代理来加速GitHub访问。因此，希望你们能够重视这个问题，并尽快采取措施解决。
我相信，有了你们的支持和可能的解决方案，我们可以改善这种情况。我强烈建议GitHub团队考虑在中国增加服务器节点或探索其他解决方案来解决这一长期存在的问题。
希望这条承载着大部分中国开发者心声的反馈不会石沉大海。我真心期待着你们的积极回应，为我们解决这一困扰已久的问题。
祝好！
Cat Bayi

**Comment by cvyl** [comment]
I am sorry for the situation you're in. Didn't China themselves make a GitHub clone named Gitee they want others to use? I could be wrong here..

**Reply by GavZheng** [reply]
Thank you very much for your feedback. I understand that Gitee is an option, but I am particularly concerned about the problems you mentioned with Gitee, especially the fact that code is reviewed without warning (both public and private repositories) and repositories are blocked for no reason and our work is ruined. Gitee has no useful functions such as Action. GitHub, on the other hand, has a reputation as a secure and reliable platform for open source development, with a thriving community of actively contributing and supporting developers. Compared to Gitee, GitHub is perfect. For these reasons, I'm inclined to keep using GitHub. Anything that can help us solve this problem and take one more "waiting for GitHub" off our to-do list, I will definitely become a committed GitHub fan.

**Comment by 54145a** [comment]
github只是偶尔能连上我们都来用了，huggingface一开始就被封死了
